subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1740,19 +1763,98 @@ int nrf_cloud_format_cell_pos_req_json(struct lte_lc_cells_info const *const inf
 	return -ENOMEM;
 }
 
-int nrf_cloud_format_cell_pos_req(struct lte_lc_cells_info const *const inf,
-	size_t inf_cnt, char **string_out)
+int nrf_cloud_format_wifi_req_json(struct wifi_scan_info const *const wifi,
+	cJSON *const req_obj_out)
 {
-	if (!inf || !inf_cnt || !string_out) {
+	if (!wifi || !req_obj_out || !wifi->ap_info || !wifi->cnt) {
+		return -EINVAL;
+	}
+
+	cJSON *wifi_obj = NULL;
+	cJSON *ap_array = NULL;
+
+	wifi_obj = cJSON_AddObjectToObjectCS(req_obj_out, NRF_CLOUD_LOCATION_JSON_KEY_WIFI);
+	ap_array = cJSON_AddArrayToObjectCS(wifi_obj, NRF_CLOUD_LOCATION_JSON_KEY_APS);
+	if (!ap_array) {
+		goto cleanup;
+	}
+
+	for (uint8_t cnt = 0; cnt < wifi->cnt; ++cnt) {
+		char str_buf[MAX(WIFI_MAC_ADDR_STR_LEN, WIFI_SSID_MAX_LEN) + 1];
+		struct wifi_scan_result const *const ap = (wifi->ap_info + cnt);
+		cJSON *ap_obj = cJSON_CreateObject();
+		int ret;
+
+		if (!cJSON_AddItemToArray(ap_array, ap_obj)) {
+			cJSON_Delete(ap_obj);
+			goto cleanup;
+		}
+
+		/* required items */
+		ret = snprintk(str_buf, sizeof(str_buf),
+			       "%02x:%02x:%02x:%02x:%02x:%02x",
+			       ap->mac[0], ap->mac[1], ap->mac[2],
+			       ap->mac[3], ap->mac[4], ap->mac[5]);
+		if ((ret != WIFI_MAC_ADDR_STR_LEN) ||
+		    json_add_str_cs(ap_obj, NRF_CLOUD_LOCATION_JSON_KEY_WIFI_MAC, str_buf)) {
+			goto cleanup;
+		}
+
+		/* optional */
```

Comment:
Does this comment refer to the memset 0, or including the SSID as a whole?

If referring to the memset specifically, I think it is needed, otherwise the string buffer won't be NULL-terminated, and cJSON might have a bad time

On a broader-scale, this code doesn't handle SSIDs with null characters in them -- And I'm not sure whether our API does either. Would it make sense to add a check for such SSIDs and skip them, rather than include, essentially, truncated SSIDs?

I think it's probably fine as it is for now, since the code still reliably produces valid JSON, but it's something we may want to write a ticket for

Reply:
optional API parameters... the only required parameter is MAC.

Reply:
Gotcha -- in that case, I recommend expanding the comment to say:

"SSID is optional for the location endpoint" or something to that effect